[TNL-9483][BB-5494] Updates to the Post UI

tecoholic opened the pull request:
### Description

Implements UI improvements for the posts view in the new design.

### Supporting information

- https://openedx.atlassian.net/browse/TNL-9483
- https://www.figma.com/file/LVFnO9oRHYiamoicZuXDAC/Discussion-page?node-id=1604%3A1338
- https://tasks.opencraft.com/browse/BB-5494

### Testing instructions

1. Checkout to the PR branch
2. Visit the app at https://localhost:2002/<course_id>/ and navigate to any of the topics to inspect the UI

#### Setting up Cohort for cohort icon testing

- Login to the LMS as a admin
- Go to the **Instructor** tab of the course used for testing
- Select **Cohorts** and Enable Cohorts
- Create a new cohort and add the test user to it
- Switch to the **Discussions** tab in the "Instructor" page
- Set **Specify whether discussion topics are divided** to `Cohorts`
- Under **Content-Specific Discussion Topics**, choose `Divide the selected content-specific discussion topics`
- Now check mark a couple of topics to test the cohort discussions
- Go to course discussions
- Go to the topic check marked for 'cohort' discussions and add a post or a comment
- Now check the Cohort icon with cohort name as a Tooltip overlay appearing at those posts.

### Other information

1. The [Paragon Icons set](https://paragon-edx.netlify.app/foundations/icons/) doesn't have the "\`Cohort` icon as provided in the Figma design, so the `Person` icon is used in it's place.

edx-webhook commented:
Thanks for the pull request, @tecoholic! I've created [OSPR-6490](https://openedx.atlassian.net/browse/OSPR-6490) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.




<!-- comment:external_pr -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

xitij2000 reviewed (approved):
:+1:

- [x] I tested this: tested on devstack
- [x] I read through the code

xitij2000 commented:
@tecoholic  Could you add tests for this please?

tecoholic commented:
@xitij2000 I have added the unit tests for the Cohort Icon display condition. 

I have introduced the same file from #72 as there wasn't any previous test file. This is something we need to keep in mind while merging.

xitij2000 commented:
@tecoholic  Please fix the merge conflict, and then this is good to go.

tecoholic commented:
@xitij2000 

1. The branch is now rebased on master and the merge conflicts resolved
2. The commits are squashed into a single commit
3. The commit message follows the standard format.

Kindly merge. :)

edx-webhook commented:
@tecoholic 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-discussions&entry.867055334=https://github.com/openedx/frontend-app-discussions/pull/69&entry.1484655318=https://github.com/tecoholic&entry.752974735=2022-02-24+10:43&entry.1917517419=2022-03-02+14:02&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
